livekit-rtc/livekit/rtc/video_frame.py

```diff
@@ -90,156 +99,346 @@ def create(owned_info: proto_video_frame.OwnedVideoFrameBuffer) \
+                         stride_y, stride_u, stride_v, chroma_width, chroma_height)
+
+    @staticmethod
+    def calc_data_size(height: int, stride_y: int, stride_u: int, stride_v: int) -> int:
+        return stride_y * height + stride_u * height + stride_v * height
 
 
 class I010Buffer(PlanarYuv16Buffer):
-    def __init__(self, owned_info: proto_video_frame.OwnedVideoFrameBuffer) -> None:
-        super().__init__(owned_info)
+    def __init__(self, data: bytearray,
+                 width: int,
+                 height: int,
+                 stride_y: int,
+                 stride_u: int,
+                 stride_v: int) -> None:
+
+        if len(data) < I010Buffer.calc_data_size(height, stride_y, stride_u, stride_v):
+            raise ValueError(
+                'buffer too small for I010 data. Expected {} bytes, got {}.'.format(
+                    I010Buffer.calc_data_size(height, stride_y, stride_u, stride_v), len(data)))
+
+        chroma_width = (width + 1) // 2
+        chroma_height = (height + 1) // 2
+        super().__init__(data, width, height, VideoFrameBufferType.I010,
+                         stride_y, stride_u, stride_v, chroma_width, chroma_height)
+
+    @staticmethod
+    def calc_data_size(height: int, stride_y: int, stride_u: int, stride_v: int) -> int:
+        return stride_y * height * 2 + stride_u * ((height + 1) // 2) * 2 + stride_v * ((height + 1) // 2) * 2
 
 
 class NV12Buffer(BiplanaraYuv8Buffer):
-    def __init__(self, owned_info: proto_video_frame.OwnedVideoFrameBuffer) -> None:
-        super().__init__(owned_info)
+    def __init__(self, data: bytearray,
+                 width: int,
+                 height: int,
+                 stride_y: int,
+                 stride_uv: int) -> None:
 
+        if len(data) < NV12Buffer.calc_data_size(height, stride_y, stride_uv):
+            raise ValueError(
+                'buffer too small for NV12 data. Expected {} bytes, got {}.'.format(
+                    NV12Buffer.calc_data_size(height, stride_y, stride_uv), len(data)))
 
-class ArgbFrame:
-    """
-    Mainly used to simplify the usage of to_argb method
-    So the users don't need to deal with ctypes
-    """
+        chroma_width = (width + 1) // 2
+        chroma_height = (height + 1) // 2
+        super().__init__(data, width, height, VideoFrameBufferType.NV12,
+                         stride_y, stride_uv, chroma_width, chroma_height)
 
+    @staticmethod
+    def calc_data_size(height: int, stride_y: int, stride_uv: int) -> int:
+        return stride_y * height + stride_uv * ((height + 1) // 2)
+
+
+class ArgbFrame:
     def __init__(self,
+                 data: Union[bytes, bytearray, memoryview],
                  format: VideoFormatType.ValueType,
                  width: int,
-                 height: int) -> None:
+                 height: int,
+                 stride: int = 0) -> None:
+
+        if stride == 0:
```

Comment:
Better way to default the stride? 0 looks confusing

Reply:
either 0 or None.. IMO a bit of comment would clear it up (i.e. when stride is not passed in, it defaults to size of a single row)